src/main/java/de/tum/in/www1/artemis/domain/participation/ProgrammingExerciseStudentParticipation.java

```diff
@@ -29,6 +29,10 @@ public class ProgrammingExerciseStudentParticipation extends StudentParticipatio
     @JsonView(QuizView.Before.class)
     private String branch;
 
+    @Column(name = "locked")
+    @JsonView(QuizView.Before.class)
+    private Boolean locked = true;
```

Comment:
```suggestion
    private boolean locked = true;
```
Can you make this a regular `boolean` here and add non-null and a default value to the database schema definition?

In some other places (e.g. ProgrammingExercise) we have gor nullable booleans where the `null` value is sometimes interpreted as `true`, sometimes as `false`, which of course has to be checked everywhere with e.g. `Booleans.TRUE.equals(...)`. This makes the code quite error-prone.

Reply:
Changed it to a regular `boolean`